core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1131,14 +1131,43 @@ private[spark] class BlockManager(
   }
 
   /**
+   * Called for pro-active replenishment of blocks lost due to executor failures
+   *
+   * @param blockId blockId being replicate
+   * @param existingReplicas existing block managers that have a replica
+   * @param maxReplicas maximum replicas needed
+   * @return
+   */
+  def replicateBlock(
+    blockId: BlockId,
+    existingReplicas: Set[BlockManagerId],
+    maxReplicas: Int): Unit = {
+    logInfo(s"Pro-actively replicating $blockId")
+    val infoForReplication = blockInfoManager.lockForReading(blockId).map { info =>
+      val data = doGetLocalBytes(blockId, info)
+      val storageLevel = StorageLevel(
+        info.level.useDisk,
```

Comment:
Minor nit, but a problem with the `StorageLevel` constructor is that it has a bunch of adjacent boolean parameters, so in such cases I'd usually prefer to name all of the parameters explicitly at the call site in order to avoid errors should these lines get permuted / to convince readers that the API is being used correctly.

Thus I'd probably write each line like `useDisk = info.level.useDisk,` etc.